src/Components/Classes/Form/index.jsx

```diff
@@ -1,105 +1,268 @@
-        <option value={'16'}>16hs</option>
-        <option value={'17'}>17hs</option>
-        <option value={'18'}>18hs</option>
-        <option value={'19'}>19hs</option>
-        <option value={'20'}>20hs</option>
-        <option value={'21'}>21hs</option>
-      </select>
-      <select
-        name="activityId"
+    <form className={styles.form} onSubmit={(e) => onSubmit(e)}>
+      <h2>{id ? 'Edit' : 'Add'}</h2>
+      {showModal && (
+        <Modal text={id ? 'Class Updated' : 'Class Added'} isOpen success onClose={reDirect} />
+      )}
+      <Select
+        value={weekDays.day}
+        placeholder={id ? singleClass.day : 'Day'}
+        onChangeSelect={(e) => onChangeInput(e)}
+        options={weekDays}
+        nameValue={'day'}
+      />
+      <Select
+        value={hours.day}
+        placeholder={id ? singleClass.hour : 'Hour'}
+        onChangeSelect={(e) => onChangeInput(e)}
+        options={hours}
+        nameValue={'hour'}
+      />
+      <Select
         value={newClass.activityId}
-        onChange={(e) => {
-          onChangeInput(e);
-        }}
-      >
-        <option disabled value="">
-          Activity
-        </option>
-        {activities.map((activity) => {
-          return (
-            <option key={activity._id} value={activity._id}>
-              {activity.name}
-            </option>
-          );
-        })}
-      </select>
-      <select
-        name="trainerId"
+        placeholder={id ? singleClass.activityId?.name : 'Activity'}
+        onChangeSelect={(e) => onChangeInput(e)}
+        options={updatedActivity}
+        nameValue={'activityId'}
+      />
+      <Select
         value={newClass.trainerId}
-        onChange={(e) => {
-          onChangeInput(e);
-        }}
-      >
-        <option disabled value="">
-          Trainer
-        </option>
-        {trainers.map((trainer) => {
-          return (
-            <option key={trainer._id} value={trainer._id}>
-              {trainer.firstName}
-            </option>
-          );
-        })}
-      </select>
-      <input
-        name="slots"
-        type="number"
-        value={newClass.slots}
-        onChange={(e) => onChangeInput(e)}
-        placeholder="Slots"
+        placeholder={id ? singleClass.trainerId?.firstName : 'Trainer'}
+        onChangeSelect={(e) => onChangeInput(e)}
+        options={updatedTrainers}
+        nameValue={'trainerId'}
+      />
+      <Input
+        nameValue={'slots'}
+        type={'text'}
+        value={newClass.slots.value}
+        onChangeInput={(e) => onChangeInput(e)}
+        placeholder={id ? singleClass.slots : 'Slots'}
       />
-      <button
-        type="submit"
-        onClick={() => {
-          readyToSendAndAdd && readyToSendAndAdd(newClass);
-          readyToSendAndEdit && readyToSendAndEdit(newClass);
-        }}
-      >
-        {text}
-      </button>
+      <Button text={id ? 'Edit' : 'Add'} submitting clickAction={sendClass} />
```

Comment:
The form should have a cancel or go back button, if you press edit or add, yo can't go back if you dont want to finish the actions.